docs/contributing/running-and-writing-acceptance-tests.md

```diff
@@ -1136,47 +1160,217 @@ The below are style-based items that _may_ be noted during review and are recomm
-  }
+#### Hardcoded Versions
+
+- [ ] __Uses Version Data Sources__: Hardcoded versions, e.g. RDS MySQL Engine Version `5.7.42`, should be removed (which means the AWS-defined default version will be used) or replaced with a list of preferred versions using a data source. Because versions change over times and version offerings vary from region to region and partition to partition, using the default version or providing a list of preferences ensures a version will be available. Depending on the situation, there are several data sources for versions, including:
+    - [`aws_rds_engine_version` data source](https://www.terraform.io/docs/providers/aws/d/rds_engine_version.html),
+    - [`aws_docdb_engine_version` data source](https://www.terraform.io/docs/providers/aws/d/docdb_engine_version.html), and
+    - [`aws_neptune_engine_version` data source](https://www.terraform.io/docs/providers/aws/d/neptune_engine_version.html).
+
+Here's an example of using `aws_rds_engine_version` and `data.aws_rds_engine_version.default.version`:
+
+```hcl
+data "aws_rds_engine_version" "default" {
+  engine = "mysql"
 }
-`, rName)
+
+data "aws_rds_orderable_db_instance" "test" {
+  engine                     = data.aws_rds_engine_version.default.engine
+  engine_version             = data.aws_rds_engine_version.default.version
+  preferred_instance_classes = ["db.t3.small", "db.t2.small", "db.t2.medium"]
+}
+
+resource "aws_db_instance" "bar" {
+  engine               = data.aws_rds_engine_version.default.engine
+  engine_version       = data.aws_rds_engine_version.default.version
+  instance_class       = data.aws_rds_orderable_db_instance.test.instance_class
+  skip_final_snapshot  = true
+  parameter_group_name = "default.${data.aws_rds_engine_version.default.parameter_group_family}"
+}
+```
+
+#### Hardcoded Instance Types
+
+- [ ] __Uses Instance Type Data Source__: Singular hardcoded instance types and classes, e.g., `t2.micro` and `db.t2.micro`, should be replaced with a list of preferences using a data source. Because offerings vary from region to region and partition to partition, providing a list of preferences dramatically improves the likelihood that one of the options will be available. Depending on the situation, there are several data sources for instance types and classes, including:
+    - [`aws_ec2_instance_type_offering` data source](https://www.terraform.io/docs/providers/aws/d/ec2_instance_type_offering.html) - Convenience functions declare configurations that are referenced with `data.aws_ec2_instance_type_offering.available` including:
+        - The `testAccAvailableEc2InstanceTypeForAvailabilityZone()` function (defined in `resource_aws_instance_test.go`) for test configurations using an EC2 Subnet which is inherently within a single Availability Zone
+        - The `testAccAvailableEc2InstanceTypeForRegion()` function (defined in `resource_aws_instance_test.go`) for test configurations that do not include specific Availability Zones
```

Comment:
We should avoid specifying file locations which will change over time and do not affect the underlying code discoverability

```suggestion
        - The `testAccAvailableEc2InstanceTypeForAvailabilityZone()` function for test configurations using an EC2 Subnet which is inherently within a single Availability Zone
        - The `testAccAvailableEc2InstanceTypeForRegion()` function for test configurations that do not include specific Availability Zones
```